### PR TITLE
fix: Add new extension failed state

### LINKED ISSUE #2209: Extension loading errors breaking external extension load
### Bug description

If one extension has misconfigured `main` script, or that script not present on filesystem for some reasons, it breaks rest of extension to load.

The code responsible for extension loading is there:
https://github.com/containers/podman-desktop/blob/23ccff4cf6bd778023d8a3543c00ed41d51a97c1/packages/main/src/plugin/extension-loader.ts#L226-L227

So if any error appears during the extension load, there are no code to handle that error, and whole sequence of loading extension is broken, so rest of extension will not load.

### Operating system

 macOS 13.2.1 Intel

### Version

0.14.1

### Steps to reproduce

Just break one of the builtin extension, by changing in package.json `main` field, to point to non existing file.
And start application, you will see that extensions that should be loaded after extension that you broke, will not load. 

### Relevant log output

```shell
(node:76301) UnhandledPromiseRejectionWarning: Error: Cannot find module '/Users/yevhen/work/redhat/podman-desktop/extensions/podman/dist/extensionaqawww.js'
Require stack:
- /Users/yevhen/work/redhat/podman-desktop/packages/main/dist/index.cjs
- /Users/yevhen/work/redhat/podman-desktop/node_modules/electron/dist/Electron.app/Contents/Resources/default_app.asar/main.js
- 
    at Module._resolveFilename (node:internal/modules/cjs/loader:1054:15)
    at n._resolveFilename (node:electron/js2c/browser_init:2:109877)
    at Module._load (node:internal/modules/cjs/loader:900:27)
    at f._load (node:electron/js2c/asar_bundle:2:13330)
    at module2._load (/Users/yevhen/work/redhat/podman-desktop/packages/main/dist/index.cjs:37643:31)
    at Module.require (node:internal/modules/cjs/loader:1120:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at ExtensionLoader.loadRuntime (/Users/yevhen/work/redhat/podman-desktop/packages/main/dist/index.cjs:38090:12)
    at ExtensionLoader.loadExtension (/Users/yevhen/work/redhat/podman-desktop/packages/main/dist/index.cjs:37797:26)
    at async Promise.all (index 5)
```
```


### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #2209

### What does this PR do?

Protect extension activation and deactivation from failure and add a new failed state

### Screenshot/screencast of this PR

image

### What issues does this PR fix or reference?

Fixes #2209 

### How to test this PR?

- Update extension activation function to throw an error
- Launch
- Go to Extensions settings page

### CODE REVIEW

**Comment by benoitf** [comment]
@jeffmaury maybe error should be displayed in the UI as well ?
So no need to go to the dev console

And send to the telemetry ( could be follow up PR)

**Comment by jeffmaury** [comment]
> @jeffmaury maybe error should be displayed in the UI as well ? So no need to go to the dev console
> 
> And send to the telemetry ( could be follow up PR)

You can see the state in the extensions page ? Do you have something else in mind ?

**Comment by benoitf** [comment]
@jeffmaury to see the root cause of the errored state
The failure, not only that it has failed

**Comment by deboer-tim** [comment]
`Errored` isn't used that frequently. Could we use `Failed` or `Failed to start` or something else?

**Comment by jeffmaury** [comment]
> @jeffmaury to see the root cause of the errored state The failure, not only that it has failed

Not sure to get it do you want the error details to be displayed on the extension page or a dialog to be shown when the error happens ?

**Comment by benoitf** [comment]
@jeffmaury on the extension page.

we can store the error in another map
```ts
this.extensionStateErrors.set(extension.id, error);
```

and if state is in error, we display the error (and we reset the field each time we start/restart the extension)

File: packages/main/src/plugin/extension-loader.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export class ExtensionLoader {
   private analyzedExtensions = new Map<string, AnalyzedExtension>();
   private watcherExtensions = new Map<string, containerDesktopAPI.FileSystemWatcher>();
   private reloadInProgressExtensions = new Map<string, boolean>();
-  private extensionState = new Map<string, string>();
+  protected extensionState = new Map<string, string>();
```
Comment (benoitf) [inline packages/main/src/plugin/extension-loader.ts:92]: not for this PR, but do we need to replace string by 'started' | 'failed' | 'stopped' ?

Comment (jeffmaury) [reply packages/main/src/plugin/extension-loader.ts:92]: Agree

**Comment by benoitf** [comment]
Hi,

FYI I tried this PR with issue https://github.com/containers/podman-desktop/issues/2451

but it doesn't say kind extension is in error state on my side
image

**Review by benoitf — APPROVED** [review]
works for me now 🎉 
image

But for example I can't start again the extension ?

**Comment by benoitf** [comment]
I should probably be able to start again the extension (probably for kind I will have the same failure but it could be possible)

**Pull request merged**